Tiltfile

```diff
@@ -1,53 +1,130 @@
+helm_resource(
+    "capi-operator",
+    "capi-operator-repo/cluster-api-operator",
+    namespace="capi-operator-system",
+    flags=["--create-namespace", "--wait"],
+    resource_deps=["capi-operator-repo", "cert-manager"],
+    labels=["CAPI"],
+)
+namespace_create("capi-system")
+k8s_yaml("./hack/manifests/core.yaml")
+k8s_resource(
+    new_name="capi-controller-manager",
+    objects=["capi-system:namespace", "cluster-api:coreprovider"],
+    resource_deps=["capi-operator"],
+    labels=["CAPI"],
 )
+if os.getenv("INSTALL_KUBEADM_PROVIDER", "true") == "true":
+    namespace_create("kubeadm-control-plane-system")
+    namespace_create("kubeadm-bootstrap-system")
+    k8s_yaml("./hack/manifests/kubeadm.yaml")
+    k8s_resource(
+        new_name="kubeadm-controller-manager",
+        objects=[
+            "kubeadm-bootstrap-system:namespace",
+            "kubeadm-control-plane-system:namespace",
+            "kubeadm:bootstrapprovider",
+            "kubeadm:controlplaneprovider",
+        ],
+        resource_deps=["capi-controller-manager"],
+        labels=["CAPI"],
+    )
 
-if os.getenv('INSTALL_K3S_PROVIDER'):
-    local_resource(
-        'capi-k3s-controller-manager',
-        cmd='clusterctl init --bootstrap k3s --control-plane k3s',
+if os.getenv("INSTALL_HELM_PROVIDER", "true") == "true":
+    namespace_create("caaph-system")
+    k8s_yaml("./hack/manifests/helm.yaml")
+    k8s_resource(
+        new_name="helm-controller-manager",
+        objects=["caaph-system:namespace", "helm:addonprovider"],
+        resource_deps=["capi-controller-manager"],
+        labels=["CAPI"],
     )
 
-if os.getenv('INSTALL_RKE2_PROVIDER'):
-    local_resource(
-        'capi-rke2-controller-manager',
-        cmd='clusterctl init --bootstrap rke2 --control-plane rke2',
+if os.getenv("INSTALL_K3S_PROVIDER", "true") == "true":
+    namespace_create("capi-k3s-control-plane-system")
+    namespace_create("capi-k3s-bootstrap-system")
+    k8s_yaml("./hack/manifests/k3s.yaml")
+    k8s_resource(
+        new_name="k3s-controller-manager",
+        objects=[
+            "capi-k3s-bootstrap-system:namespace",
+            "capi-k3s-control-plane-system:namespace",
+            "k3s:bootstrapprovider",
+            "k3s:controlplaneprovider",
+        ],
+        resource_deps=["capi-controller-manager"],
+        labels=["CAPI"],
+    )
+
+if os.getenv("INSTALL_RKE2_PROVIDER", "true") == "true":
+    namespace_create("rke2-control-plane-system")
+    namespace_create("rke2-bootstrap-system")
+    k8s_yaml("./hack/manifests/rke2.yaml")
+    k8s_resource(
+        new_name="capi-rke2-controller-manager",
+        objects=[
+            "rke2-bootstrap-system:namespace",
+            "rke2-control-plane-system:namespace",
+            "rke2:bootstrapprovider",
+            "rke2:controlplaneprovider",
+        ],
+        resource_deps=["capi-controller-manager"],
+        labels=["CAPI"],
```

Comment:
Question: Do we want to install k3s and rke2 provider by default? I thought we would only want to install it when the env var is set to true

Reply:
oops, good catch, fixed